framework/src/base/MooseApp.C

```diff
@@ -106,6 +106,8 @@ validParams<MooseApp>()
                                    "--check-input",
                                    false,
                                    "Check the input file (i.e. requires -i <filename>) and quit.");
+  params.addCommandLineParam<bool>(
+      "regurgitate", "--regurgitate", false, "Re-create the input file.");
```

Comment:
Funny - but we already have a flag called "--show-input" that needs some love. Maybe we should just <s>regurgitate</s> reuse that

Reply:
They do different things, --show-input shows the exact input. This shows the exploded version.

Reply:
Ha! It's never shown anything close to the exact output. It was supposed to show the  interpreted output but it's always been pretty terrible which is probably why it's just not used at all. What I'm saying is that we really don't need both or at least we could deprecate the former unless somebody wants to make it work with the new HIT parser. Exploded and interpreted maybe, but the latter is currently very broken. maybe "show-expanded" would be a better name.